Break out Installation into top-level doc section

alexkrolick opened the pull request:
- No more jumping over the installation section with links pointing to the "first" Quick Start page
- Get rid of some of that custom footer code that hardcodes "installation" as a special case
- Installation is important, may as well have its own section 😛 

![image](https://user-images.githubusercontent.com/1571667/34234031-1f57b038-e595-11e7-9a76-fbdf26459035.png)

bvaughn commented on content/docs/installation-try-react.md:
```diff
@@ -0,0 +1,21 @@
+  - "docs/environments.html"
+---
+
+React is flexible and can be used in a variety of projects. You can create new apps with it, but you can also gradually introduce it into an existing codebase without doing a rewrite.
```
I'm not sure this lead paragraph (and the below header) make sense anymore, given what this page has been remade into.

Maybe we should rename the page title to something like "Try it on CodePen" and leave out this paragraph entirely?

alexkrolick replied:
Sure, although, how about adding more than Codepen? The Codesandbox template for React is pretty mature and quite popular, and we could link out to other content like the editable examples on the home page.

bvaughn replied:
Sure. That'd be totally fine. CodeSandbox comes to mind as another excellent solution.

bvaughn commented on content/docs/installation-try-react.md:
```diff
@@ -0,0 +1,21 @@
+  - "docs/tooling-integration.html"
+  - "docs/package-management.html"
+  - "docs/language-tooling.html"
+  - "docs/environments.html"
```
The new Installation pages should have prev/next frontmatter links 😄

alexkrolick replied:
See #451 - the lack of sync between the titles and the ids causes the next/prev to appear as "Installation New Apps", etc

bvaughn replied:
Not sure I understand. The next/prev titles are currently generated based on the URL you put in the next/prev frontmatter (so it's something we control, even if it's not synced).

I think the new section should still have next/prev links to feel more consistent with the others.

alexkrolick replied:
```
//  installation-existing-app.md
id: installation-existing-app
title: Add React to an Existing Application
```

```
// elsewhere
next: installation-existing-app.html
```

Results in:

![image](https://user-images.githubusercontent.com/1571667/34284507-01889780-e67f-11e7-8dde-18819918e54c.png)

Instead of:

![image](https://user-images.githubusercontent.com/1571667/34284525-220d518a-e67f-11e7-847d-77cbf02e16b6.png)

This is because of #451 where the url is parsed instead of looking up the title in the TOC data structure. I am planning on fixing that and adding next/prev once it is done, but **I can certainly adjust the filenames for now so that it looks correct**.

bvaughn replied:
> so it's something we control, even if it's not synced

You chose to name the new file `installation-existing-app.md`. So I'm just saying, before #451 is resolved, we could work around this issue by just naming it `add-react-to-an-existing-app.md` instead. 😄

alexkrolick commented:
> It feels odd though for "Installation" to come after "Quick Start".

Yeah, a little, but it does feel a bit weird that lots of links to Quick Start appear to be skipping a section if that's where you land from the home page.

bvaughn commented:
I think that's okay, or at least preferable to this. I wonder if people actively _look_ for installation- they'll expect it to come first (and may miss it entirely if it's collapsed below an expanded list of items.)

gaearon commented on content/docs/try-react.md:
```diff
@@ -0,0 +1,22 @@
+
+If you're just interested in playing around with React, you can use CodePen. Try starting from [this Hello World example code](http://codepen.io/gaearon/pen/rrpgNB?editors=0010). You don't need to install anything; you can just modify the code and see if it works.
+
+If you prefer to use your own text editor, you can also [download this HTML file](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html), edit it, and open it from the local filesystem in your browser. It does a slow runtime code transformation, so don't use it in production.
```
I think we should add a section here that points to the next two sections and explains in which case you'll want which. It's too easy to miss them otherwise.

alexkrolick replied:
Working on this 😛

alexkrolick replied:
Pushed a change - added "Local Development" section to the Try React page

alexkrolick commented:
@bvaughn 
- Moved Installation to top
- Added prev/next by changing filenames
- Added Codesandbox

@gaearon 
- Reorganized and added links to the other install guides

gaearon commented on content/docs/try-react.md:
```diff
@@ -0,0 +1,33 @@
+
+## Install React
+
+- **[Add React to a New App](/docs/add-react-to-a-new-app.html)** Create a new app with a starter kit like [Create React App].(http://github.com/facebookincubator/create-react-app)
```
The link Markdown is broken

gaearon commented on content/docs/try-react.md:
```diff
@@ -0,0 +1,33 @@
+
+If you prefer to use your own text editor, you can also [download this HTML file](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html), edit it, and open it from the local filesystem in your browser. It does a slow runtime code transformation, so don't use it in production.
+
+## Install React
```
This makes it sound like it's part of *this* page. IMO this makes it worse. I was thinking something that more obviously implies it's an optional *next* step, not something you have to do.

gaearon commented on content/docs/try-react.md:
```diff
@@ -0,0 +1,33 @@
+## Install React
+
+- **[Add React to a New App](/docs/add-react-to-a-new-app.html)** Create a new app with a starter kit like [Create React App].(http://github.com/facebookincubator/create-react-app)
+- **[Add React to an Existing App](/docs/add-react-to-a-new-app.html)** _Advanced configuration._ Use React with a build system like Webpack.
```
I don't think this description is very helpful.

alexkrolick replied:
I'm not sure the dichotomy is correct here - "Add React to A New App" links to CRA primarily, and "Existing App" is all about build systems and CDN links. It makes sense to promote CRA as the starting point, but I think it is confusing to talk about the next page as only for existing apps. It's more of an advanced guide, or maybe a bundling/build guide. The CDN links could also probably be more prominent, with the caveat that they aren't very useful without Babel.

What do you think?

gaearon replied:
I understand what you're saying. My point is just that 

>_Advanced configuration._ Use React with a build system like Webpack.

is not very meaningful to someone visiting the page for the first time. What advanced configuration? What is webpack? Does the first link not use webpack? etc I'm open to other phrasings of the same idea, this sentence just read as confusing to me.

>I think it is confusing to talk about the next page as only for existing apps

In a way, yes. But on the other hand it provides a clear entry point for each use case. If you're ready to configure things on your own you already know enough that you don't need either of these sections.

alexkrolick replied:
Just changed the description to "Add React to a build system or larger app." --- how does that sound? Can iterate on the naming of the page later.

Also added a "Next Steps" header above the Quick Start and Install sections to make it clear that those aren't part of Trying React, but ideas for what to do next if you're set up.

gaearon commented on content/docs/try-react.md:
```diff
@@ -0,0 +1,37 @@
+
+### Quick Start
+
+Head over to the [Quick Start](/docs/hello-world.html) section to try some examples.
```
I think it's a bit misleading to call QS section that (since it doesn't have that many useful "examples").

Maybe it would be better to direct people either to Tutorial (for those looking for a hands-on practical introduction) or to QS (but frame it as a step-by-step introduction to React concepts).

alexkrolick replied:
Linking to the Tutorial is a great idea.

gaearon commented:
I'm really liking these changes. Left one last nit I have.

alexkrolick commented:
- Pulled CDN links to the top level
- Added link to tutorial page

Care to take one more pass?

😄

gaearon commented on content/docs/try-react.md:
```diff
@@ -0,0 +1,39 @@
+  - "docs/environments.html"
+---
+
+Try React online or set up your local dev environment
```
"development environment"

Also let's end with period like we do elsewhere?

gaearon commented on content/docs/try-react.md:
```diff
@@ -0,0 +1,39 @@
+
+If you're just interested in playing around with React, you can use an online code playground. Try a Hello World template on [CodePen](codepen://hello-world) or [CodeSandbox](https://codesandbox.io/s).
+
+## HTML Template
```
Maybe "Minimal HTML Template" to make it clearer what this is about

gaearon commented on content/docs/try-react.md:
```diff
@@ -0,0 +1,39 @@
+
+- Try the [Tutorial](/tutorial/tutorial.html) for a hands-on practical example.
+
+### Install React
```
Again, this sounds like it's a required step because of the title.

How about: "Powerful Development Environment" or "Complete Development Environment". Maybe also change the sentence to explain when/why you'd want that.

The goal is to explain this step takes time but gives you a great developer experience and a production-ready setup with linting, testing, and optimizations built-in. We have a chance to preempt "omg why is it 100mb" knee-jerk responses here by being explicit that it's not just for 'hello world' but more IDE-like.

alexkrolick replied:
👍

gaearon commented on content/docs/add-react-to-a-new-app.md:
```diff
@@ -0,0 +1,42 @@
+next: add-react-to-an-existing-app.html
+---
+
+The easiest way to get started on a new React project is by using a starter kit.
```
Related to https://github.com/reactjs/reactjs.org/pull/452/files#r158572397: maybe it's worth pointing people looking for something lightweight to "Try React" from this page. I'd love if we could make it clearer that this is a fully-featured setup that isn't *required* for using React but covers common needs for a *single-page application* including linting, testing, optimizations, etc.

alexkrolick replied:
👍  Added a note here

gaearon commented:
I think this is great! Hopefully nothing breaks 😄 
Thank you so much for reorganizing this!
